### PR TITLE
PMM-9299 fix: add signout_redirect_url and increase timeout

### PR DESCRIPTION
**Pull request by jprukner**
[PMM-9299](https://jira.percona.com/browse/PMM-9299)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/2186

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Review by JiriCtvrtka — APPROVED** [review]

File: services/supervisord/supervisord.go [hunk 1]
```diff
@@ -697,6 +708,7 @@ command =
         {{- if .PerconaSSODetails}}
         cfg:default.server.domain="{{ .PMMServerAddress }}"
         cfg:default.auth.generic_oauth.enabled=true
+        cfg:default.auth.signout_redirect_url="https://{{ .IssuerDomain }}/login/signout?fromURI=https://{{ .PMMServerAddress }}/graph/login"
```
Comment (atymchuk) [inline services/supervisord/supervisord.go:711]: Well, problem is we should support domains like http://localhost. Therefore, can we gran the protocol (http/https) from the PMM's url? My problem is PMM always assumes https< which is what breaks my PMM deployment on http://localhost. Someone tagged me today for the exact same problem.

Comment (jprukner) [reply services/supervisord/supervisord.go:711]: @atymchuk I think that is limitation of Okta. We have to use https. Am I right @maxkondr ?

Comment (maxkondr) [reply services/supervisord/supervisord.go:711]: Okta rejects application domain with scheme != https.

Comment (maxkondr) [reply services/supervisord/supervisord.go:711]: please see my comment https://jira.percona.com/browse/PMM-9287?focusedCommentId=292441&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-292441

**Review by maxkondr — APPROVED** [review]

File: services/platform/platform.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func New(db *reform.DB, supervisord supervisordService) (*Service, error) {
 	return &s, nil
 }
 
-const platformAPITimeout = 10 * time.Second
+const platformAPITimeout = 30 * time.Second
```
Comment (maxkondr) [inline services/platform/platform.go:73]: can we move it to some ENV and have a default value if it is not set?

Comment (maxkondr) [reply services/platform/platform.go:73]: because it really depends on user's network

File: utils/envvars/parser.go [hunk 3]
```diff
@@ -36,8 +36,10 @@ const (
 	envSaaSHost     = "PERCONA_TEST_SAAS_HOST"
 	envPublicKey    = "PERCONA_TEST_CHECKS_PUBLIC_KEY"
 	// TODO REMOVE PERCONA_TEST_DBAAS IN FUTURE RELEASES.
-	envTestDbaas   = "PERCONA_TEST_DBAAS"
-	envEnableDbaas = "ENABLE_DBAAS"
+	envTestDbaas              = "PERCONA_TEST_DBAAS"
+	envEnableDbaas            = "ENABLE_DBAAS"
+	envPlatfromAPITimeout     = "PERCONA_TEST_PLATFORM_API_TIMEOUT"
```
Comment (maxkondr) [inline utils/envvars/parser.go:41]:
```suggestion
	envPlatfromAPITimeout     = "PERCONA_PLATFORM_API_TIMEOUT"
```

this is not for testing but may be used in production as well by users

File: services/platform/platform.go [hunk 4]
```diff
@@ -88,7 +90,7 @@ func (s *Service) Connect(ctx context.Context, req *platformpb.ConnectRequest) (
 	}
 	pmmServerURL := fmt.Sprintf("https://%s/graph", settings.PMMPublicAddress)
 
-	nCtx, cancel := context.WithTimeout(ctx, platformAPITimeout)
+	nCtx, cancel := context.WithTimeout(ctx, s.platformAPITimeout)
 	defer cancel()
 
 	ssoParams, err := s.connect(nCtx, &connectPMMParams{
```
Comment (maxkondr) [inline services/platform/platform.go:96]: is it possible to apply it to all outgoing requests but not just connect?

**Review by maxkondr — CHANGES_REQUESTED** [review]

File: utils/envvars/parser.go (hunk 3 above)
Comment (jprukner) [reply utils/envvars/parser.go]: Done

File: services/platform/platform.go (hunk 4 above)
Comment (jprukner) [reply services/platform/platform.go:96]: Done.

Comment (maxkondr) [reply services/platform/platform.go:96]: is this change still needed? I see the timeout is applied to http client in general

Comment (jprukner) [reply services/platform/platform.go]: Indeed not needed. Removed.

**Review by maxkondr — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

**Pull request merged**